drivers/pinctrl/CMakeLists.txt

```diff
@@ -21,6 +21,7 @@ zephyr_library_sources_ifdef(CONFIG_PINCTRL_IMX pinctrl_imx.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_SIFIVE pinctrl_sifive.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_NXP_IOCON pinctrl_lpc_iocon.c)
 zephyr_library_sources_ifdef(CONFIG_PINCTRL_CC13XX_CC26XX pinctrl_cc13xx_cc26xx.c)
+zephyr_library_sources_ifdef(CONFIG_PINCTRL_CC23X0 pinctrl_cc23x0.c)
```

Comment:
please squash pinctrl changes

Reply:
Are you talking about these 3 commits ?
```
ab61cf258af35fc07d5bbde3eca2739e4cebcad7 dt-bindings: pinctrl: cc23x0: Add definitions
319c9a847cc49cf0c47510f767d36bd45ffbad42 drivers: pinctrl: Add support for cc23x0 pinctrl
2d7d295c6628e9fdcd46530c3fabac1796fc30ad soc: ti: cc23x0: Add pinctrl support
```

I'm a bit confused because there was also this previous request:
https://github.com/zephyrproject-rtos/zephyr/pull/76441/commits/2d7d295c6628e9fdcd46530c3fabac1796fc30ad#r1762648373

Reply:
combo commits mean mixing unrelated stuff to the same commit.

Reply:
OK, thank you. So I'll squash the 3 commits I mentioned above.